torch/_functorch/_aot_autograd/input_output_analysis.py

```diff
@@ -347,7 +349,7 @@ def _tensors_definitely_do_not_overlap(x, y):
     return False
 
 
-def compute_overlapping_inputs(fwd_inputs, aliased_input_indices):
+def compute_overlapping_inputs(aot_config, fwd_inputs, aliased_input_indices):
     max_aliased_inps_w_dyn_shapes = (
         config._max_aliased_inputs_with_dynamic_shapes_enabled
     )
```

Comment:
In theory if this PR fixes the issue properly, then we can eventually kill the code below that needs to potentially error when there are dynamic shapes (not actionable in this PR though)